### PR TITLE
Coverage for Test Criteria (tested using Windows AMD-64)

### PR DESCRIPTION
**Pull request by kevinmstapleton**
created flag to find coverage for test criteria and created makefile (in the form of a batch script) for Windows users

### CODE REVIEW

File: cmd/atrutil/atrutil_main.go [hunk 1]
```diff
@@ -218,6 +220,126 @@ func FillInToolPathDefaults() {
...
+		r.LazyQuotes = true
+		r.Comment = '#'
+		r.FieldsPerRecord = -1 // no validation on num columns per row
+
+		row, err := r.Read()
+
+		if err == nil && strings.HasPrefix(row[0], "T") {
+
+			//search the mapping to see if the test exists inside of the full list of atomic tests
+			if atomicMap[row[0]] != nil {
```
Comment (amalone-scwx) [inline cmd/atrutil/atrutil_main.go]: you will need to do something like 
```
  criteria = utils.AtomicTestCriteriaNew(row[0], row[1], row[2], row[3])
  for _,entry := range atomicMap[criteria.Technique] {
     if len(criteria.TestGuid) > 0 && strings.HasPrefix(entry.GUID,criteria.TestGuid) {
        // match
        break
     }
     if criteria.TestIndex > 0 && criteria.TestIndex == entry.TestIndex {
        // match
        break
     }
   }

**Comment by amalone-scwx** [comment]
it looks like the flagPlatform change I merged caused a conflict.  Can you please pull upstream and resolve?  Sorry about that.

**Comment by kevinmstapleton** [comment]
No problem, an easy fix

**Comment by kevinmstapleton** [comment]
Just pushed those changes and is ready for re-review

File: cmd/atrutil/atrutil_main.go [hunk 2]
```diff
@@ -220,6 +222,140 @@ func FillInToolPathDefaults() {
...
+
+			cur := utils.AtomicTestCriteriaNew(row[0], row[1], row[2], row[3])
+
+			for _, entry := range atomicMap[cur.Technique] {
+				if len(cur.TestGuid) > 0 && strings.HasPrefix(entry.TestGuid, cur.TestGuid) {
+					criteria += 1
+					break
+				}
+				fmt.Print("Current Test Index: ", cur.TestIndex, "\nEntry TestIndex: ", entry.TestIndex, "\n")
+				fmt.Print("Current Test Name: \n", cur.TestName, "\n")
```
Comment (amalone-scwx) [inline cmd/atrutil/atrutil_main.go]: you probably want to hide these debug logging statements in a `if gVerbose {}` block

File: cmd/atrutil/atrutil_main.go [hunk 3]
```diff
@@ -220,6 +222,140 @@ func FillInToolPathDefaults() {
...
+				entryIndex, err := strconv.ParseUint(entry.TestIndex, 10, 64)
+
+				if err != nil {
+					fmt.Println("failed to parse entry index: ", err)
+					break
+				}
```
Comment (amalone-scwx) [inline cmd/atrutil/atrutil_main.go]: You can use the `ToUInt()` convenience function inside this file.

Comment (kevinmstapleton) [reply cmd/atrutil/atrutil_main.go]: Thanks for pointing that out! Updates are pushed (_hopefully last one!_) and I made the output a lot less verbose by default including the suggestion above

**Pull request merged**